stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -93,12 +101,22 @@ internal class ChannelLogic(
     private val mutableState: ChannelMutableState,
     private val chatDomainImpl: ChatDomainImpl,
     private val attachmentUrlValidator: AttachmentUrlValidator = AttachmentUrlValidator(),
+    messageSendingServiceFactory: MessageSendingServiceFactory = MessageSendingServiceFactory(),
 ) : QueryChannelListener, ChannelMarkReadListener, GetMessageListener, HideChannelListener {
 
     private val logger = ChatLogger.get("Query channel request")
 
     private var lastMarkReadEvent: Date? by mutableState::lastMarkReadEvent
 
+    private val messageSendingService: MessageSendingService =
```

Comment:
Please extract creation logic from ChannelLogic, this class should not instantiate its dependencies.

Reply:
🤔  I don't think that ChannelLogic should be aware of `MessageSendingService`. This logic belongs to `SendMessageListenerImpl`.